test/unit/org/apache/cassandra/net/ConnectionUtils.java

```diff
@@ -240,6 +247,27 @@ private void doCheck(FailCheck testAndFailCheck)
         }
     }
 
+    private static void longCheck(Runnable assertion, long timeout, TimeUnit timeUnit)
+    {
+        long start = System.currentTimeMillis();
```

Comment:
`System.currentTimeMillis` is free to go backwards (its not monotonic, this is system time) so its normally best to use `.nanoTime()` (CPU time).